crates/trigger-http/src/handler.rs

```diff
@@ -210,29 +212,40 @@ impl HttpHandlerExecutor {
             None => Handler::Latest(Proxy::new(&mut store, &instance)?),
         };
 
-        let handle = task::spawn(async move {
-            let result = match handler {
-                Handler::Latest(proxy) => {
-                    proxy
-                        .wasi_http_incoming_handler()
-                        .call_handle(&mut store, request, response)
-                        .await
-                }
-                Handler::Handler2023_10_18(proxy) => {
-                    proxy.call_handle(&mut store, request, response).await
-                }
-                Handler::Handler2023_11_10(proxy) => {
-                    proxy.call_handle(&mut store, request, response).await
-                }
-            };
-
-            tracing::trace!(
-                "wasi-http memory consumed: {}",
-                store.as_ref().data().memory_consumed()
-            );
-
-            result
-        });
+        let span = tracing::debug_span!("execute_wasi");
+        let handle = task::spawn(
+            async move {
+                let result = match handler {
+                    Handler::Latest(proxy) => {
+                        proxy
+                            .wasi_http_incoming_handler()
+                            .call_handle(&mut store, request, response)
+                            .instrument(span)
+                            .await
+                    }
+                    Handler::Handler2023_10_18(proxy) => {
+                        proxy
+                            .call_handle(&mut store, request, response)
+                            .instrument(span)
+                            .await
+                    }
+                    Handler::Handler2023_11_10(proxy) => {
+                        proxy
+                            .call_handle(&mut store, request, response)
+                            .instrument(span)
+                            .await
+                    }
+                };
+
+                tracing::trace!(
+                    "wasi-http memory consumed: {}",
+                    store.as_ref().data().memory_consumed()
+                );
+
+                result
+            }
+            .in_current_span(),
```

Comment:
Does this work properly when the debug level is disabled?

Reply:
Yes. The trace is still properly constructed in Jaeger even when the debug spans aren't set off to Jaeger.

Reply:
I'm more concerned about `tracing` losing track of the "current" span if it drops out of scope before this spawned task completes. It would be good to verify with an app that does streaming responses like `examples/wasi-http-rust-streaming-outgoing-body`.

Reply:
Karthik and I spent some time playing around with the examples to make sure this was working. Most of the examples were broken, but despite that we didn't observe any issues with tracing losing track of the "current" span. Let me know if you want further investigation here.